web/src/message_flags.js

```diff
@@ -1,17 +1,32 @@
 import _ from "lodash";
 
 import * as channel from "./channel";
+import {localstorage} from "./localstorage";
 import * as starred_messages from "./starred_messages";
 
 export function send_flag_update_for_messages(msg_ids, flag, op) {
-    channel.post({
-        url: "/json/messages/flags",
-        data: {
-            messages: JSON.stringify(msg_ids),
-            flag,
-            op,
-        },
-    });
+    const ls = localstorage();
+    channel
+        .post({
+            url: "/json/messages/flags",
+            data: {
+                messages: JSON.stringify(msg_ids),
+                flag,
+                op,
+            },
+        })
+        ?.fail(() => {
```

Comment:
This is not how you check for failures in the `channel` module; you pass an `error` hook. The fact that you wrote this code this way suggests to me that you didn't test this. You should be testing, say by stopping your server so that all requests fail.

Reply:
@timabbott , I tested it with ```fail``` section, and it indeed worked. The reference for this section : [docs](https://api.jquery.com/deferred.fail/).  I also found out that the ```error``` hook was deprecated with ```fail``` for ajax requests.

Reply:
Ahh, OK. The `channel` module isn't designed to support that with its error wrapper logic; though maybe we should change it, but I think if we care we might make `channel` use `fail` internally.

Reply:
Ya, actually that time when I wrote the code, I didn't have the idea of how it was done.... I'll update it.